File: 02_Stack/02_Min_Stack/155_Min_Stack.java
class MinStack {

    Stack<Integer> s = new Stack<>();
    Stack<Integer> min_s = new Stack<>();

    public MinStack() {
        
    }
    
    public void push(int val) {
        s.push(val);

        if (min_s.empty() || val <= min_s.peek()){
            min_s.push(val);
        }
        else {
            min_s.push(min_s.peek());
        }
    }
    
    public void pop() {
        s.pop();
        min_s.pop();
    }
    
    public int top() {
        return s.peek();
    }
    
    public int getMin() {
        return min_s.peek();
    }
}

File: 02_Stack/02_Min_Stack/155_Min_Stack.cpp
class MinStack {
    stack<int> s;
    stack<int> min_s;
public:
    MinStack() {
    }
    
    void push(int val) {
        s.push(val);
        if (min_s.size() == 0 || val <= min_s.top()){
            min_s.push(val);
        }
        else {
            min_s.push(min_s.top());
        }
    
    }
    
    void pop() {
        s.pop();
        min_s.pop();
    }
    
    int top() {
        return s.top();
    }
    
    int getMin() {
        return min_s.top();
        
    }
};